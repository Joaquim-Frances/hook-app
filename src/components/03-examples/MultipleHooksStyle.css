body{
    padding: 70px;
}

.title{
    border: 1px solid grey;
    border-radius: 10px;
    box-shadow: 0px 0px 10px 2px rgb(148, 148, 148) ;
    color: purple;
    margin: 20px;
    padding: 20px;
    width: fit-content;
}