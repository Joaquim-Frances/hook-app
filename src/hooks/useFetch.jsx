import { useEffect, useState } from "react"



export const useFetch = ( url, id ) => {
    
    const [state, setState] = useState({ data: null, loading: true, error: null});

    useEffect( ()=>{

        fetch(url)
            .then( respuesta => respuesta.json())
            .then( data => {

                setState({
                    loading: false,
                    error: null,
                    data
                })

            })

    }, [url, id])

    return state;

}