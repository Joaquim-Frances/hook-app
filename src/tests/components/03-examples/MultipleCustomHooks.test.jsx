import React from 'react';
import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks"
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Testing MultipleCustomHook component', () => {

    beforeEach (() => {
        
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {},
        });
      
    })

    test('should display correctly', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null,
        })

        const wrapper = shallow( <MultipleCustomHooks />);
        
        expect( wrapper ).toMatchSnapshot();
        
    })
    
    test('should display information when it have it', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'Quim',
                quote: 'Nama Gana, Nama Gana, Gana Gana Nam Nam!',
            }],
            loading: false,
            error: null,
        })
        
        const wrapper = shallow( <MultipleCustomHooks />);
        
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-3').text().trim()).toBe('Nama Gana, Nama Gana, Gana Gana Nam Nam!');
        expect(wrapper.find('footer').text().trim()).toBe('Quim');
        
        
    })
    
    
})
