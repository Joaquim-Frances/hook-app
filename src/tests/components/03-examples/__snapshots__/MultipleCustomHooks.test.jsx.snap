// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MultipleCustomHook component should display correctly 1`] = `
<div>
  <h1
    className="title"
  >
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
     Mas frases!
  </button>
</div>
`;
